src/app/AttributeAccessInterface.h

```diff
@@ -414,6 +414,34 @@ class AttributeAccessInterface
      */
     virtual CHIP_ERROR Write(const ConcreteDataAttributePath & aPath, AttributeValueDecoder & aDecoder) { return CHIP_NO_ERROR; }
 
+    /**
+     * Indicates the start of a series of list operations. This function will be called before the first Write operation of a series
+     * of consequence attribute data of the same attribute.
+     *
+     * 1) This function won't be called if the user writes a null value to a list attribute.
```

Comment:
Why not?

Reply:
Because we can only infer the operation from the data received.
For replace all, we can infer the operation from the data
For append, we can infer the operation from the path

For nullable list, if a null value is received, we don't have enough infomation for it.

Reply:
That assumes we only use data from over-the-wire type information provided by TLV, but that need not be true. Every attribute has a corresponding descriptor that can be queried which contains its type information. That should be used to retrieve this information.